engine/script/src/script.cpp

```diff
@@ -698,6 +698,23 @@ namespace dmScript
         return type_hash;
     }
 
+
+    uint32_t RegisterUserTypeLocal(lua_State* L, const char* name, const luaL_reg meta[])
```

Comment:
I opted for a separate function to keep the RegisterUserType() clean.
This function allows us to register an "object style" syntax for the b2d object.
I felt that that was cleaner, and easier to browse the api between the official Box2D docs and also to understand how our module is working.